cpp/tests/test_ide_seir.cpp

```diff
@@ -0,0 +1,80 @@
+/* 
+* Copyright (C) 2020-2021 German Aerospace Center (DLR-SC)
+*
+* Authors: Lena Ploetzke
+*
+* Contact: Martin J. Kuehn <Martin.Kuehn@DLR.de>
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+#include "load_test_data.h"
+#include "ide_seir/ide_seir.h"
+#include "ide_seir/ide_seir.cpp"
+#include "ide_seir/parameters.h"
+#include "memilio/math/eigen.h"
+#include "memilio/utils/time_series.h"
+#include "memilio/epidemiology/uncertain_matrix.h"
+#include <gtest/gtest.h>
+
+class IdeSeirModelTest : public testing::Test
+{
+protected:
+    virtual void SetUp()
+    {
+        using Vec = mio::TimeSeries<double>::Vector;
+
+        int N     = 810000;
+        double dt = 0.1;
+        mio::TimeSeries<double> result(1);
+        result.add_time_point<Eigen::VectorXd>(-15.0, Vec::Constant(1, N * 0.95));
+        while (result.get_last_time() < 0) {
+            result.add_time_point(result.get_last_time() + dt,
+                                  Vec::Constant(1, (double)result.get_last_value()[0] + result.get_last_time()));
+        }
+
+        model = new mio::iseir::IdeSeirModel(std::move(result), dt, N);
+
+        model->m_parameters.set<mio::iseir::LatencyTime>(3.3);
+        model->m_parameters.set<mio::iseir::InfectiousTime>(8.2);
+        model->m_parameters.set<mio::iseir::TransmissionRisk>(0.015);
```

Comment:
Do you need model to be a pointer for TearDown or why did you not use a reference?

Reply:
I thought this is needed for the TearDown, yes.